core/src/muxing.rs

```diff
@@ -91,14 +91,19 @@ pub trait StreamMuxer {
     /// is ready to be polled, similar to the API of `Stream::poll()`.
     /// Only the latest task that was used to call this method may be notified.
     ///
+    /// Implementors may rely on this method being called soon after the registered
+    /// task has been notified. It is thus possible to use this method to perform
+    /// background work, such as processing incoming packets and polling timers.
+    ///
     /// An error can be generated if the connection has been closed.
     fn poll_event(&self, cx: &mut Context) -> Poll<Result<StreamMuxerEvent<Self::Substream>, Self::Error>>;
 
     /// Opens a new outgoing substream, and produces the equivalent to a future that will be
     /// resolved when it becomes available.
     ///
     /// The API of `OutboundSubstream` is totally opaque, and the object can only be interfaced
-    /// through the methods on the `StreamMuxer` trait.
+    /// through the methods on the `StreamMuxer` trait. Interacting with `OutboundSubstream`
+    /// directly has undefined results.
```

Comment:
Again, what is the motivation for bringing up undefined behaviour here?

Reply:
It isn’t actually UB, but it might corrupt the internal state of the muxer and cause future misbehavior.

Reply:
Since `OutboundSubstream` is an unconstrained type parameter it is up to the `StreamMuxer` impl to define what interactions are possible. The trait can not and should not anticipate what interactions lead to "misbehavior".

Reply:
Fixed.